src/c_api.cpp

```diff
@@ -841,6 +852,31 @@ int LGBM_BoosterLoadModelFromString(
   API_END();
 }
 
+int LGBM_BoosterCreateForPythonProto(
```

Comment:
@guolinke I use two functions: one to create booster, one to pass pandas category from c++ to python. Is there a better idea?